usmqe/usmscheduler.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf8 -*-
+"""
+Pytest plugin for scheduling tasks on nodes.
+.. moduleauthor:: Filip Balak <fbalak@redhat.com>
+"""
+import usmqe.usmssh as usmssh
+import datetime
+import random
+import string
+
+
+class Scheduler(object):
+    """
+    Wrapper around `usmssh` module that handles scheduling tasks on multiple
+    nodes simultaneously.
+    """
+
+    def __init__(self, nodes):
+        """
+        Args:
+            nodes (list): list of nodes used for command scheduling
+        """
+        self.nodes = nodes
+        self.ssh = usmssh.get_ssh()
+
+    def run_at(self, command, time=None):
+        """
+        Schedule a job on all machines with `at` command.
+
+        Args:
+            command (str): Command to be executed.
+            time (time): Time to execute. Time has to be in `at` compatible
+                format. If None is provided then all commands are executed
+                in next minute.
+        """
+        if not time:
+            time = datetime.datetime.now() + datetime.timedelta(minutes=1)
+            time = time.strftime("%H:%M")
+        files = self.create_job_file(command)
+        for node in self.nodes:
+            time_command = "at {0} -f {1} ".format(time, files[node])
```

Comment:
One note: it would be good to convert the timestamp from local timezone to UTC and pass it that way to at. This way, the code would work no matter what timezone is configured on local usmqe server and remote machines.

Reply:
Another point is: use seconds in the timestamp as well. It could happen that there is very little time between the time the job was scheduled and actual execution, eg:

```
[10:26:57,499] [ INFO     ] ssh:: Executing 'at 10:27 -f /tmp/schedulertask_7i4z8g ' on mbukatov-usm2-gl1.example.com
```